mutable_tree.go

```diff
@@ -229,8 +232,8 @@ func (tree *MutableTree) recursiveRemove(node *Node, key []byte) ([]byte, *Node,
 	}
 	newNode.calcHeightAndSize(tree.ImmutableTree)
 	newNode, balanceOrphaned := tree.balance(newNode)
```

Comment:
Why can't we update `balance` function to take in an orphan list as an argument and append there?

Reply:
good point, changed.